### ISSUE TITLE
Add tag support

### ISSUE THREAD

**Issue by joente** [SiriDB/siridb-server#74]
Currently we can create dynamic groups based on a regular expression for grouping series but there is no way to manually add/remove series to a group.

Sometimes this is useful so we should implement tagging of series.

Syntax:

Tags should be created automatically as soon as a series is tagged.
```
alter series `a-group-or-even-a-tag-is-allowed`, 'by-name', /or_by_regex/ where type == float tag `my-tag`
```
>In case a group exists with the same name, an error must me returned. No error should be raised if 
>a series has already a tag assigned.

To add series to an existing tag is done with the same syntax:
```
alter series 'my-series' tag `my-tag`
```

Remove a tag should be possible:
```
alter series 'my-series' untag `my-tag`
```
>No error is raised if the series did not have the tag, an error should be raised when the tag does not
>exist.

Tags should be list-able:
```
list tags name, series where name ~ 'cpu' and series > 10
```

Tags should be countable:
```
count tags where name ~ 'foo'
```

Tags are removed as soon as a tag is not used anymore:
```
alter series `my-tag` untag `my-tag`
```

**Comment by joente** [comment]
The [tags](https://github.com/transceptor-technology/siridb-server/tree/tags) branch has working code for tagging and untagging series.

The following issues need to be solved for a final implementation:
  * What to do when selecting series from a group or tag that does not exist. Since tags are not know by pools which have no series for a specific tag, we can't tell if a tag exists or not and therefore can't tell if a group or tag exists.
  * How to extend a cluster with a new replica? One solution is to make a copy from the '/tags' directory and start synchronizing the tags after all series are synchronized and then continue the 'normal' changes to tags. Another solution which is probably better is to create an initial sync file for tags. And then synchronize the tags like we do with series.
  * How to extend a cluster with a new pool? One solution is to flag a migrated series before it is dropped on the 'original' server. The taks (thread) which is responsible for actually cleaning up the series from existing tags could create a package with tag information and send this package to the new server. This does not guarantee correct flagging in case of a server failure but aside from that, the solution should work.

**Comment by joente** [comment]
By allowing "empty" tags we can solve most of the issues. This does require a `drop tag` syntax otherwise existing tags will not be removed.

While synchronizing the series, we can perform a "tag" lookup and synchronize the tags for each series straight after the series data is send.

**Comment by joente** [comment]
Solved in SiriDB version 2.0.28-alpha-0
